Document user personas (#773)
conda-incubator/conda-store: docusaurus-docs/community/design/user-personas.md

---
description: Describes current and potential conda-store users`
---

# User groups and personas

The personas used for informing user research studies and future improvements to conda-store.

## User groups 🏘️

At a high-level, there are two key user groups:

- **General user:** Create, edit, use environments for their day-to-day work

> - **Admin-type user:** In addition to the general use-case mentioned above, admin-type users also **setup** conda-store for their team, create various namespaces, add/update users and their permissions, set default packages/versions for their team, etc.

Author: Good catch, I think that was left behind from discussions/notes. I'll update it.

## User personas 🙋🏽‍♀️

### User: Alia

| Role | Reports to | User group|
|-------|-------|-------|
Staff Data Scientist | Head of Data Science | General |

#### User story

I want to quickly create a stable and reproducible environment for analyzing data and creating dashboards that I can share with my team to verify and build on my work.

#### Tools they need to do their job

- Libraries - data reader/writer (arrow), data processing (pandas), numerical computing (numpy), ML (xgboost / pytorch), dataviz and dashboarding (mpl / streamlit), job schedulers (Airflow)
- IDE - JupyterLab/Notebook (hence, extensions)

> - Platform - Local computer, org JupyterHub on Cloud (Nebari?)

Author: ```suggestion
- Platform - Local computer, org JupyterHub on Cloud (potentially Nebari)
```

- Misc - Git, GitHub, conda

#### Journey with conda-store

- Discovery:
    - Team already uses it
    - Team leader sets it up for everyone
- Onboarding:
    - A team member gives them a walkthrough
    - Documented Tutorials
    - Intuitiveness of the UI
- General Use:
    - Env creation from GUI, YAML, Lockfile
    - Version control
    - Using the env in Jupyter Notebook (local / cloud)
    - Ability to use any package/version out there
    - Access to special conda channels

>     - Envs for GPU-powered data analysis and modelling

Author: US English :)

- Collaboration:
    - Share env with colleagues with a URL (Part of 1-2 shared namespaces)
    - Fork/Copy envs between namespaces
    - Download artifacts like docker images for use in other (prod?) machines
- Troubleshooting:
    - Understand errors through the error message displayed
    - Look at conda-store’s documentation & GH issues/discussions, search for the error
    - Contact team-members or internal support
    - Open an issue/discussion on GH

#### Pain points or biggest challenges

> - Envs need to be compliant with company standards

Author: ```suggestion
- Environments need to be compliant with company standards, for example: approved conda channels and approved package versions
```

> - Envs need to be compliant with company standards

Author: added more info :)

> - Build envs with libraries from internal mirrors

Reviewer: This is in contrast with "ability to use any package/version out there" https://github.com/conda-incubator/conda-store/pull/773/files#diff-99d5523c0e67c61630b4cfb694883c9282df0074dcd29f29078be74bdb1efc2fR48

Author: I don't think so? We meant this as: internal mirrors are one of the various places we'd want to install packages from, right?

- Ensure stable environments that are quickly reproducible by colleagues and can be used by Ops teams for deployment (often on similar machines and operating-systems)

#### Core needs

- Intuitive and fast env creation & sharing
- Promise of stability, security, and compliance

### User: Dani

| Role | Reports to | User group|
|-------|-------|-------|
Research Scientist and Coordinator | PI | Admin |

#### User story

> As a research coordinator, I manage the logistics and admin-tasks of multiple research projects, ensuring that the tools/dependencies used are accessible, reproducible and standardised across projects.

Reviewer: ```suggestion
As a research coordinator, I manage the logistics and admin-tasks of multiple research projects, ensuring that the tools/dependencies used are accessible, reproducible and standardized across projects.
```

#### Tools they need to do their job

> In addition to Alia’s tools, cloud/hpc tooling

Author: This user would interface with GCP for example, and use the gcloud CLI. They may also dig into the k8s logs and such. I'll add some clarification. :)

#### Journey with conda-store

- Discovery:
    - OSS Community, conferences, peers
    - Tries out conda-store locally and finds it useful
- Onboarding:
    - Documentation, self-exploration
- General Use:
    - Setup, deployment using the conda-store docs, within current Lab infra
    - Add team members, create default namespaces and environments, referring to the docs
    - Onboard team members to the tool with demonstrations

>     - Track and limit resource utilisation

Reviewer: ```suggestion
    - Track and limit resource utilization
```

#### Core needs

- Setting up & managing packaging infra for the lab, potentially on an HPC system as as non-devops professional
- Visibility into groups’ package requirements and resources used
- Sharing environments widely, considering different operating systems and infrastructures.

### User: Emma

| Role | Reports to | User group|
|-------|-------|-------|
Freelance data scientist | - | General |

#### User story

I want to manage my local conda environments, so that I can be more efficient in my day-to-day work

#### Tools they need to do their job

Same as Alia, but primarily for individual work

#### Journey with conda-store

- Discovery
    - Internet, community spaces, conferences
- Onboarding
    - Docs
- General Use
    - Install and setup conda-store locally
    - Create envs in my personal namespace
    - Use the envs in Jupyter Notebooks
- Collaboration
    - Share relevant artifacts (lockfiles?) with clients
- Troubleshooting
    - Docs, issue tracker

#### Pain points or biggest challenges

- Sometimes conda environments break while working, adding overhead
- Environments are not always reproducible when I share them

> - Using shared artifacts w/ the application should be as easy as possible (think installers/executables)

Reviewer: ```suggestion
- Using shared artifacts with the application should be as easy as possible (think installers/executables)
```
